libs/features/mas/web-components/src/hydrate.js

```diff
@@ -84,24 +84,24 @@ export async function hydrate(fragmentData, merchCard) {
         );
     }
 
-    if (fragment.backgroundImage) {
-        switch (variant) {
-            case 'ccd-slice':
-              if (aemFragmentMapping.backgroundImage) {
-                merchCard.append(
-                  createTag(
-                    aemFragmentMapping.backgroundImage.tag,
-                    { slot: aemFragmentMapping.backgroundImage.slot },
-                    `<img loading="lazy" src="${fragment.backgroundImage}" />`
-                  )
-                );
-              }
+    switch (variant) {
+        case 'ccd-slice':
```

Comment:
I know it was already there but don't think we should have variant specific logic in hydrate.js.

It should be driven by `aemFragmentMapping` which provides the variant specific configuration.
If a metadata is missing, please add in the corresponding variant file.
Can you by this occasion make it more generic?

btw this file will heavy change in my PR:https://github.com/adobecom/milo/pull/3117/files#diff-212e6466053994809d44f48d318154b20f4d124480d1df3281c83a454553bc10

Reply:
Sure, I will then wait for your PR to be merged and make the changes.